include/uk/plat/smp.h

```diff
@@ -33,8 +33,15 @@
 #ifndef __PLAT_CMN_SMP_H__
 #define __PLAT_CMN_SMP_H__
 
+#include <inttypes.h>
+struct pcpu {
+	uint32_t	pc_cpuid;	/* This cpu number */
+};
+
 #define	MAXCPU		8 /* hard limitation for CPU number */
 extern int cpu_possible_map[MAXCPU];
+struct pcpu __pcpu[MAXCPU];
+struct pcpu *pcpup = &__pcpu[0];
```

Comment:
Do we need these two vectors: cpu_possible_map[] and pcpu? I think the logic could be implemented only using the cpu IDs vector.....